tm2/pkg/sdk/params/keeper.go

```diff
@@ -0,0 +1,157 @@
+package params
+
+import (
+	"log/slog"
+	"strings"
+
+	"github.com/gnolang/gno/tm2/pkg/amino"
+	"github.com/gnolang/gno/tm2/pkg/sdk"
+	"github.com/gnolang/gno/tm2/pkg/store"
+)
+
+const (
+	ModuleName = "params"
+	StoreKey   = ModuleName
+)
+
+type ParamsKeeperI interface {
+	GetString(ctx sdk.Context, key string, ptr *string)
+	GetInt64(ctx sdk.Context, key string, ptr *int64)
+	GetUint64(ctx sdk.Context, key string, ptr *uint64)
+	GetBool(ctx sdk.Context, key string, ptr *bool)
+	GetBytes(ctx sdk.Context, key string, ptr *[]byte)
+
+	SetString(ctx sdk.Context, key string, value string)
+	SetInt64(ctx sdk.Context, key string, value int64)
+	SetUint64(ctx sdk.Context, key string, value uint64)
+	SetBool(ctx sdk.Context, key string, value bool)
+	SetBytes(ctx sdk.Context, key string, value []byte)
+
+	Has(ctx sdk.Context, key string) bool
+	GetRaw(ctx sdk.Context, key string) []byte
+
+	// XXX: ListKeys?
+}
+
+var _ ParamsKeeperI = ParamsKeeper{}
+
+// global paramstore Keeper.
+type ParamsKeeper struct {
+	key    store.StoreKey
+	prefix string
+}
+
+// NewParamsKeeper returns a new ParamsKeeper.
+func NewParamsKeeper(key store.StoreKey, prefix string) ParamsKeeper {
+	return ParamsKeeper{
+		key:    key,
+		prefix: prefix,
+	}
+}
+
+// Logger returns a module-specific logger.
+// XXX: why do we expose this?
+func (pk ParamsKeeper) Logger(ctx sdk.Context) *slog.Logger {
+	return ctx.Logger().With("module", ModuleName)
+}
+
+func (pk ParamsKeeper) Has(ctx sdk.Context, key string) bool {
+	stor := ctx.Store(pk.key)
+	return stor.Has([]byte(key))
+}
+
+func (pk ParamsKeeper) GetRaw(ctx sdk.Context, key string) []byte {
+	stor := ctx.Store(pk.key)
+	return stor.Get([]byte(key))
+}
+
+func (pk ParamsKeeper) GetString(ctx sdk.Context, key string, ptr *string) {
```

Comment:
why do you see us needing to support all of these types?

Reply:
The generic tm2 module is designed to support the top five types while ensuring strong type safety. 

In practice, we could rely solely on the `Bytes` type and expect users to handle the encoding. However, after discussing it with Jae, we agreed that providing these built-in types for convenience seems like a good approach.